src/m365/planner/commands/task/task-add.ts

```diff
@@ -0,0 +1,390 @@
+  
+        if (args.options.description) {
+          return this.getTaskDetailsEtag(taskId)
+            .then(etag => {
+              const requestOptionsTaskDetails: any = {
+                url: `${this.resource}/v1.0/planner/tasks/${taskId}/details`,
+                headers: {
+                  'accept': 'application/json;odata.metadata=none',
+                  'If-Match': etag,
+                  'Prefer': 'return=representation'
+                },
+                responseType: 'json',
+                data: {
+                  description: args.options.description
+                }
+              };
+      
+              return request.patch(requestOptionsTaskDetails);              
+            })
+            .then(taskDetails => {
+              return { ...newTask, ...taskDetails as PlannerTaskDetails };
+            });
+        } 
+        else {
+          return newTask;
+        }  
+      })
+      .then((res: any): void => {
+        logger.log(res);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getTaskDetailsEtag(taskId: string): Promise<string> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/tasks/${encodeURIComponent(taskId)}/details`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get(requestOptions)
+      .then((response: any) => {
+        const etag: string | undefined = response ? response['@odata.etag'] : undefined;
+
+        if (!etag) {
+          return Promise.reject(`Error fetching task details`);
+        }
+
+        return Promise.resolve(etag);
+      });
+  }
+
+  private generateUserAssignments(args: CommandArgs): Promise<{ [userId: string]: PlannerAssignment; }> {
+    const assignments: { [userId: string]: PlannerAssignment; } = {};
+
+    if (args.options.assignedToUserNames) {
+      return this.getUserIds(args.options.assignedToUserNames)
+        .then((userIds) => {
+          userIds.map(x => assignments[x] = {
+            orderHint: " !"
+          });
+
+          return Promise.resolve(assignments);
+        });
+    }
+    else if (args.options.assignedToUserIds) {
+      args.options.assignedToUserIds.split(',').map(x => assignments[x] = {
+        orderHint: " !"
+      });
+
+      return Promise.resolve(assignments);
+    } 
+    else {
+      return Promise.resolve(assignments);
+    }
+  }
+
+  private getBucketId(args: CommandArgs, planId: string): Promise<string> {
+    if (args.options.bucketId) {
+      return Promise.resolve(args.options.bucketId);
+    }
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/plans/${planId}/buckets`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: { id: string; name: string; }[] }>(requestOptions)
```

Comment:
Let's use the Bucket type from the Graph typings instead of defining our own type